test/e2e/context.ts

```diff
@@ -172,6 +172,7 @@ export class TestContext {
     await this.waitForTrue(
       async () => (this.mainProcess ? !(await this.mainProcess.isRunning()) : false),
       () => 'Timeout: Waiting for app to die',
+      15_000,
```

Comment:
Is there a particular reason why we chose this number?

Reply:
I changed this when the test originally tested against an unresolvable DSN. It would take forever to timeout and fail. I decided to test only against the server returning 429/500 because waiting for the connection to timeout was too flakey in the tests so this lengthy timeout can be reverted.